hugr/src/hugr/serialize/test.rs

```diff
@@ -349,79 +367,34 @@ fn serialize_types_roundtrip() {
     assert_eq!(ser_roundtrip(&t), t);
 }
 
-#[rstest]
-#[case(BOOL_T)]
-#[case(USIZE_T)]
-#[case(INT_TYPES[2].clone())]
-#[case(Type::new_alias(crate::ops::AliasDecl::new("t", TypeBound::Any)))]
-#[case(Type::new_var_use(2, TypeBound::Copyable))]
-#[case(Type::new_tuple(type_row![BOOL_T,QB_T]))]
-#[case(Type::new_sum([type_row![BOOL_T,QB_T], type_row![Type::new_unit_sum(4)]]))]
-#[case(Type::new_function(FunctionType::new_endo(type_row![QB_T,BOOL_T,USIZE_T])))]
-fn roundtrip_type(#[case] typ: Type) {
-    check_testing_roundtrip(typ);
-}
-
-#[rstest]
-#[case(SumType::new_unary(2))]
-#[case(SumType::new([type_row![USIZE_T, QB_T], type_row![]]))]
-fn roundtrip_sumtype(#[case] sum_type: SumType) {
-    check_testing_roundtrip(sum_type);
-}
-
-#[rstest]
-#[case(Value::unit())]
-#[case(Value::true_val())]
-#[case(Value::unit_sum(3,5).unwrap())]
-#[case(Value::extension(ConstF64::new(-1.5)))]
-#[case(Value::extension(ConstF64::new(0.0)))]
-#[case(Value::extension(ConstF64::new(-0.0)))]
-// These cases fail
-// #[case(Value::extension(ConstF64::new(std::f64::NAN)))]
-// #[case(Value::extension(ConstF64::new(std::f64::INFINITY)))]
-// #[case(Value::extension(ConstF64::new(std::f64::NEG_INFINITY)))]
-#[case(Value::extension(ConstF64::new(std::f64::MIN_POSITIVE)))]
-#[case(Value::sum(1,[Value::extension(ConstInt::new_u(2,1).unwrap())], SumType::new([vec![], vec![INT_TYPES[2].clone()]])).unwrap())]
-#[case(Value::tuple([Value::false_val(), Value::extension(ConstInt::new_s(2,1).unwrap())]))]
-#[case(Value::function(crate::builder::test::simple_dfg_hugr()).unwrap())]
-fn roundtrip_value(#[case] value: Value) {
-    check_testing_roundtrip(value);
-}
+#[cfg(feature = "proptest")]
+mod proptest {
+    use super::super::NodeSer;
+    use super::check_testing_roundtrip;
+    use crate::extension::ExtensionSet;
+    use crate::ops::{OpType, Value};
+    use crate::types::{PolyFuncType, Type};
+    use proptest::prelude::*;
+
+    proptest! {
+        #[test]
+        fn prop_roundtrip_type(t:  Type) {
+            check_testing_roundtrip(t)
```

Comment:
Yay :). I was a bit confused when I looked at the wealth of methods with `roundtrip` in the name (it might help to rename `ser_roundtrip_validate` to `ser_schema_roundtrip` as it does not call `validate`), but this one looks to do what we want :-). You could remove the `allow_unused` on its declaration now!